libs/src/utils/promiseFight.ts

```diff
@@ -0,0 +1,24 @@
+/**
+ * Given an array of promises, it returns the first resolved promise as soon as it finishes
+ * @param promises
+ * @param captureErrored optional capture errored promises
+ * @return A promise that resolves with the first promise that resolves
+ */
+export async function promiseFight (promises: Array<Promise<unknown>>, captureErrored = false) {
```

Comment:
jsyk, this method is essentially a polyfill for Promise.any. Curious if you know the latest & greatest there?

Reply:
i don't, but can def look into